applications/sintering/include/pf-applications/sintering/driver.h

```diff
@@ -2305,6 +2305,18 @@ namespace Sintering
                     save_all_blocks,
                     params.grain_cut_off_tolerance);
 
+                // Check Courant condition
+                if (params.advection_data.enable &&
+                    params.advection_data.check_courant)
+                  {
+                    advection_operator.evaluate_forces(solution,
+                                                       advection_mechanism);
```

Comment:
It appears here because we need to obtain the advection velocities and for that we need to compute the Wang forces. If the Courant check was performed at the end of the step, then there is no need to do it since the forces are already computed (at each call of `nl_residual()`).